`transmute` doctest type error

https://github.com/CHERIoT-Platform/cheri-rust/blob/6c8a6a78c3a2a0454881a555b7741141039740e6/library/core/src/intrinsics/mod.rs#L676-L686

Manifesting as a failing doctest (ptr !== usize)

>error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
   --> library/core/src/intrinsics/mod.rs:681:5
    |
681 |     std::mem::transmute::<&i32, usize>(ptr)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: source type: `&i32` (64 bits)
    = note: target type: `usize` (32 bits)

--- a/library/core/src/intrinsics/mod.rs
+++ b/library/core/src/intrinsics/mod.rs
@@ -677,9 +677,9 @@ pub const fn forget<T: ?Sized>(_: T);
 /// assert_eq!(num, 0x78563412);
 /// ```
 ///
-/// Turning a pointer into a `usize`:
+/// Turning a pointer into a `usize` (which isn't an option on CHERI targets anyway):
 ///
-/// ```no_run
+/// ```no_run,ignore-cheri
 /// let ptr = &0;
 /// let ptr_num_transmute = unsafe {
 ///     std::mem::transmute::<&i32, usize>(ptr)

